CoreQualityControlView/src/au/gov/asd/tac/constellation/views/qualitycontrol/QualityControlViewPane.java

```diff
@@ -680,46 +677,34 @@ private static void readSerializedRulePriorities() {
     public static Map<QualityControlRule, QualityCategory> getPriorities() {
         if (MapUtils.isEmpty(rulePriorities)) {
             rulePriorities = new HashMap<>();
-            for (final QualityControlRule rule : getLookup().lookupAll(QualityControlRule.class)) {
+            for (final QualityControlRule rule : Lookup.getDefault().lookupAll(QualityControlRule.class)) {
                 rulePriorities.put(rule, rule.getCategory(0));
             }
         }
         return rulePriorities;
     }
 
-    /**
-     * Lazily instantiates the lookup object
-     *
-     * @return the cached lookup object
-     */
-    public static Lookup getLookup() {
```

Comment:
I believe @arcturus2 mentioned that in the past there has been slowness issues with always fetching the lookup. I think this was added as a way to store the lookup object - Keen to hear if @arcturus2 thoughts are still the same or if it's a non-issue

Reply:
You've got a good memory @aldebaran30701. So there is definitely a performance issue making `Lookup` calls like `Lookup.getDefault().lookup(Something.class)` but `Lookup.getDefault()` should be fine as its just asking for the singleton version of `Lookup` which is OK.

Just to double check I profiled the 2 methods where `Lookup` was being called in this branch and can confirm its fine.

![image](https://user-images.githubusercontent.com/39325530/117104220-1a81b900-adbf-11eb-9cfd-fb8cdb1d35d5.png)

Reply:
Thanks for the detailed response! Continue as you were then :D